reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -246,12 +252,20 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
           }
         } ~
         (pathEnd & put & entity(as[SignedPayload[TargetsRole]])) { signedPayload =>
+          // get them before they get removed from the DB
+          val previousTargetItemsF = targetItemRepo.findFor(repoId)
+
           extractRoleChecksumHeader { checksum =>
             onSuccess(offlineSignedRoleStorage.saveTargetRole(namespace, signedPayload, repoId, checksum)) {
```

Comment:
I think a more robust solution would be to delete the targets in the analogous position to where we store them, if possible.

Reply:
What do you mean? Put the deletion code next to the storing code?

Reply:
I mean somewhere inside `saveTargetRole` we save the file to s3/storage, we should delete it inside somewhere in `saveTargetRole` I think.

Reply:
I don't think `saveTargetRole` saves the package files to storage.

Reply:
Sorry, my mistake. Ok then

Reply:
Updated.